server/repository/src/database_settings.rs

```diff
@@ -72,8 +73,19 @@ impl DatabaseSettings {
         }
     }
 
+    pub fn database_path(&self) -> String {
+        let result = match &self.database_path {
+            Some(path) => {
+                std::fs::create_dir_all(path).expect("failed to create database dir");
+                format!("{}/{}", path, self.connection_string())
+            }
+            None => self.connection_string(),
+        };
+        return result;
+    }
+
     pub fn connection_string_without_db(&self) -> String {
-        self.connection_string()
+        self.database_path()
```

Comment:
Does this work with postgres?

Reply:
Sorry I see it's behind the `#[cfg(all(not(feature = "postgres"), not(feature = "memory")))]` so I think it should be ok.

Reply:
but you will need to add this to the postgres `get_test_db_settings`
```
error[E0063]: missing field `database_path` in initializer of `DatabaseSettings`
  --> repository/src/test_db/postgres.rs:11:5
   |
11 |     DatabaseSettings {
   |     ^^^^^^^^^^^^^^^^ missing `database_path`
```

Try this PR with `cargo test --features=postgres` and probably start up the app in postgres mode as well for good measure :)

Reply:
This `connection_string_without_db` is only used for the postgres feature right? could you please try to remove it for sqlite and memory? think it should work without it...